### ISSUE TITLE
Etas and changestats

### ISSUE THREAD

**Issue by mbojan** [statnet/ergm#9]
This is a child of #6 

I am having troubles with understanding what exactly `ergm.eta` returns and how does it square with the output of `ergmMPLE`. For example:

```r
data(sampson)
gest <- ergm(
  samplike ~ edges + gwesp(decay=0.5, fixed=FALSE),
  control=control.ergm(MCMLE.maxit=1)
)
etas <- ergm.eta(coef(gest), gest$etamap)
a <- ergmMPLE(
  samplike ~ edges + gwesp(decay=0.5, fixed=FALSE),
  output="array"
)
```

Now `etas` has length 17 (!) while the `predictors` array has only two layers for edges and GWESP terms. Does `etas` contain entries for the whole ESP distribution?

**Comment by mbojan** [comment]
Am I correct that eta contains geometrical weights for the individual components of the ESP distribution? It seems that's what calling `gest$etamap$curved[[1]]$map()` does...

**Comment by krivit** [comment]
Are you running this against `master` or against `dev`? Their behaviour may be different in this respect.

**Comment by mbojan** [comment]
That's Cran version

**Comment by krivit** [comment]
Ah, I see. This is because `ergmMPLE` is passing `initialfit` to the terms, forcing them to act like non-curved. (Since `dev` has full curved MPLE support, that is not the case there.)

**Comment by mbojan** [comment]
Ok, I'll switch to `dev` then.

**Comment by mbojan** [comment]
... which I guess also means that we move the discussion to `-private`

**Comment by krivit** [comment]
I don't think it uses any unpublished APIs, so it's fine.

**Comment by mbojan** [comment]
Errrr.... It make sense to implement it with a current state of funs. You just wrote that ergm.eta has been updated on dev, which is in private. I will need to make PRs to private. Or we can talk in public and push in private, so to speak...

**Comment by mbojan** [comment]
So I rebased against `dev` and the code from the inital OP gives me errors

```
d> data(sampson)
d> gest <- ergm(
+   samplike ~ edges + gwesp(decay=0.5, fixed=FALSE),
+   control=control.ergm(MCMLE.maxit=1)
+ )
Starting maximum pseudolikelihood estimation (MPLE):
Evaluating the predictor and response matrix.
Maximizing the pseudolikelihood.
Finished MPLE.
Starting Monte Carlo maximum likelihood estimation (MCMLE):
Iteration 1 of at most 1:
Error in NextMethod("[") : argument is missing, with no default
In addition: Warning message:
 Show Traceback
 
 Rerun with Debug
 Error in NextMethod("[") : argument is missing, with no default 
```

this uses statnet.common from GH and

```
R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.4 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=pl_PL.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=pl_PL.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=pl_PL.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=pl_PL.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] magrittr_1.5   ergm_4.0-3888  network_1.13.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.16             xml2_1.2.0               roxygen2_6.0.1           devtools_1.13.5         
 [5] MASS_7.3-49              lattice_0.20-35          R6_2.2.2                 rlang_0.2.0             
 [9] stringr_1.3.0            tools_3.4.4              parallel_3.4.4           grid_3.4.4              
[13] lpSolve_5.6.13           coda_0.19-1              withr_2.1.2              commonmark_1.4          
[17] yaml_2.1.18              digest_0.6.15            Matrix_1.2-12            testthat_2.0.0          
[21] trust_0.1-7              memoise_1.1.0            robustbase_0.92-8        statnet.common_4.1.0-161
[25] stringi_1.1.7            compiler_3.4.4           DEoptimR_1.0-8          
```

**Comment by mbojan** [comment]
@krivit I'm running into even more errors:

```
n <- 4
net <- network.initialize(n)
net[1,2] <- net[2,3] <- 1
ergmMPLE(net ~ edges + offset(edges), output="array")
Error in eta[ec[ec > 0]] <- theta[ec > 0] : replacement has length zero
In addition: Warning message:
In is.na(theta.offset[m$etamap$offsettheta]) :
```

I think `dev` is too unstable. We are rushing too much. On the other hand, developing it on top of `master` having upcoming changes to `ergm.eta` etc. from `dev` does not make sense either. Perhaps this should wait for the relevant codebase to stabilize.

**Comment by krivit** [comment]
Fixed in `dev` ( ffc5cc089e3 ).

**Comment by krivit** [comment]
Since this seems to be inactive and the issues raised have been fixed, I am closing this. Feel free to reopen if you want to continue.
